Mathlib/Analysis/Normed/Unbundled/AlgNormOfGalois.lean

```diff
@@ -0,0 +1,143 @@
+* [S. Bosch, U. Güntzer, R. Remmert, *Non-Archimedean Analysis*][bosch-guntzer-remmert]
+
+## Tags
+
+algNorm_of_auto, algNorm_of_galois, norm, nonarchimedean
+-/
+
+
+
+
+open scoped NNReal
+
+noncomputable section
+
+variable {K : Type*} [NormedField K] {L : Type*} [Field L] [Algebra K L]
+  (h_fin : FiniteDimensional K L)
+section algNorm_of_auto
+
+/-- Given a normed field `K`, a finite algebraic extension `L/K` and `σ : L ≃ₐ[K] L`, the function
+`L → ℝ` sending `x : L` to `‖ σ x ‖`, where `‖ ⬝ ‖` is any power-multiplicative algebra norm on `L`
+extending the norm on `K`, is an algebra norm on `K`. -/
+def algNorm_of_auto (hna : IsNonarchimedean (norm : K → ℝ)) (σ : L ≃ₐ[K] L) :
+    AlgebraNorm K L where
+  toFun x     := Classical.choose (exists_nonarchimedean_pow_mul_seminorm_of_finiteDimensional
+    h_fin hna) (σ x)
+  map_zero'   := by simp [map_eq_zero_iff_eq_zero, EmbeddingLike.map_eq_zero_iff]
+  add_le' x y := by simp [map_add σ, map_add_le_add]
+  neg' x      := by simp [map_neg σ, map_neg_eq_map]
+  mul_le' x y := by simp [map_mul σ, map_mul_le_mul]
+  smul' x y   := by simp [map_smul σ, map_smul_eq_mul]
+  eq_zero_of_map_eq_zero' x hx := EmbeddingLike.map_eq_zero_iff.mp (eq_zero_of_map_eq_zero _ hx)
+
+theorem algNorm_of_auto_apply (σ : L ≃ₐ[K] L) (hna : IsNonarchimedean (norm : K → ℝ)) (x : L) :
+    algNorm_of_auto h_fin hna σ x =
+      Classical.choose (exists_nonarchimedean_pow_mul_seminorm_of_finiteDimensional h_fin hna)
+        (σ x) := rfl
+
+/-- The algebra norm `algNorm_of_auto` is power-multiplicative. -/
+theorem isPowMul_algNorm_of_auto (σ : L ≃ₐ[K] L) (hna : IsNonarchimedean (norm : K → ℝ)) :
+    IsPowMul (algNorm_of_auto h_fin hna σ) := by
+  intro x n hn
+  simp only [algNorm_of_auto_apply, map_pow σ x n]
+  exact (Classical.choose_spec (exists_nonarchimedean_pow_mul_seminorm_of_finiteDimensional
+    h_fin hna)).1 _ hn
+
+/-- The algebra norm `algNorm_of_auto` is nonarchimedean. -/
+theorem isNonarchimedean_algNorm_of_auto (σ : L ≃ₐ[K] L) (hna : IsNonarchimedean (norm : K → ℝ)) :
+    IsNonarchimedean (algNorm_of_auto h_fin hna σ) := by
+  intro x y
+  simp only [algNorm_of_auto_apply, map_add σ]
+  exact (Classical.choose_spec (exists_nonarchimedean_pow_mul_seminorm_of_finiteDimensional
+    h_fin hna)).2.2 _ _
+
+/-- The algebra norm `algNorm_of_auto` extends the norm on `K`. -/
+theorem algNorm_of_auto_extends (σ : L ≃ₐ[K] L) (hna : IsNonarchimedean (norm : K → ℝ)) (x : K) :
+    (algNorm_of_auto h_fin hna σ) ((algebraMap K L) x) = ‖x‖ := by
+  simp only [algNorm_of_auto_apply, AlgEquiv.commutes]
+  exact (Classical.choose_spec (exists_nonarchimedean_pow_mul_seminorm_of_finiteDimensional
+    h_fin hna)).2.1 _
+
+end algNorm_of_auto
+
+section algNorm_of_galois
+
+/-- The function `L → ℝ` sending `x : L` to the maximum of `algNorm_of_auto h_fin hna σ` over
+  all `σ : L ≃ₐ[K] L` is an algebra norm on `L`. -/
+def algNorm_of_galois (hna : IsNonarchimedean (norm : K → ℝ)) : AlgebraNorm K L where
```

Comment:
I don't think we can have `galois` in the name because there's no assumption that the extension is Galois. What about `IsNonarchimedean.algNormOfFiniteDimensional`? Can you face something of this length? I feel like this is only an intermediate construction so maybe the long name is OK?

Reply:
What do you think of `algNormOfAut`, to refer to the automorphism group of L/K in the name?

Reply:
I am not completely keen on it but I don't really have any better ideas. You're not getting from from an Aut, maybe we should ask for advice. Modulo this naming issue this PR is fine.

Reply:
Since there were no further suggestions or complaints, I have renamed this to `IsUltrametricDist.invariantExtension`.